matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/terms/DefaultTermsService.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.matrix.android.internal.session.terms
+
+import dagger.Lazy
+import im.vector.matrix.android.api.MatrixCallback
+import im.vector.matrix.android.api.session.events.model.toModel
+import im.vector.matrix.android.api.session.terms.GetTermsResponse
+import im.vector.matrix.android.api.session.terms.TermsService
+import im.vector.matrix.android.api.util.Cancelable
+import im.vector.matrix.android.internal.di.Unauthenticated
+import im.vector.matrix.android.internal.network.NetworkConstants
+import im.vector.matrix.android.internal.network.RetrofitFactory
+import im.vector.matrix.android.internal.network.executeRequest
+import im.vector.matrix.android.internal.session.identity.IdentityAuthAPI
+import im.vector.matrix.android.internal.session.identity.IdentityRegisterTask
+import im.vector.matrix.android.internal.session.identity.todelete.AccountDataDataSource
+import im.vector.matrix.android.internal.session.openid.GetOpenIdTokenTask
+import im.vector.matrix.android.internal.session.sync.model.accountdata.AcceptedTermsContent
+import im.vector.matrix.android.internal.session.sync.model.accountdata.UserAccountData
+import im.vector.matrix.android.internal.session.user.accountdata.UpdateUserAccountDataTask
+import im.vector.matrix.android.internal.task.launchToCallback
+import im.vector.matrix.android.internal.util.MatrixCoroutineDispatchers
+import kotlinx.coroutines.GlobalScope
+import okhttp3.OkHttpClient
+import javax.inject.Inject
+
+internal class DefaultTermsService @Inject constructor(
+        @Unauthenticated
+        private val unauthenticatedOkHttpClient: Lazy<OkHttpClient>,
+        private val accountDataDataSource: AccountDataDataSource,
+        private val termsAPI: TermsAPI,
+        private val retrofitFactory: RetrofitFactory,
+        private val getOpenIdTokenTask: GetOpenIdTokenTask,
+        private val identityRegisterTask: IdentityRegisterTask,
+        private val updateUserAccountDataTask: UpdateUserAccountDataTask,
+        private val coroutineDispatchers: MatrixCoroutineDispatchers
+) : TermsService {
+    override fun getTerms(serviceType: TermsService.ServiceType,
+                          baseUrl: String,
+                          callback: MatrixCallback<GetTermsResponse>): Cancelable {
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
```

Comment:
Ohh.. i see now that lanchToCallBack catches all

Reply:
:-)